src/bitwarden_workflow_linter/rules/permissions_exist.py

```diff
@@ -26,18 +27,24 @@ def __init__(
         lint_level: Optional[LintLevels] = LintLevels.NONE,
     ) -> None:
         self.message = (
-            "A top-level permissions section must be configured in the workflow."
+            "All workflows must specify permissions on either the workflow level or on all jobs"
         )
         self.on_fail = lint_level
         self.compatibility = [Workflow]
         self.settings = settings
 
-    def permissions_exist(self, obj: Workflow) -> bool:
-        if obj.permissions is None:
+    def permissions_exist_on_workflow(self, workflow: Workflow) -> bool:
+        if workflow.permissions is None:
             return False
         return True
 
+    def permissions_exist_on_jobs(self, jobs: list[Job]) -> bool:
+        for job in jobs:
+            if job.permissions is None:
+                return False
+        return True
```

Comment:
This looks absolute and will fail if permission is only specified on the workflow level. Do we want to enforce permission on job levels?

Reply:
Oh, I talked too soon, that check is done in the `fn` method